dvc/repo/metrics/show.py

```diff
@@ -66,30 +67,24 @@ def _read_typed_metric(typ, xpath, fd):
     return ret
 
 
-def _read_metric(path, typ=None, xpath=None, branch=None):
-    ret = None
-    if not os.path.exists(path):
-        return ret
+def _read_metric(fd, typ=None, xpath=None, rel_path=None, branch=None):
 
     typ = typ.lower().strip() if typ else typ
-    xpath = xpath.strip() if xpath else xpath
-    try:
-        with open(path, "r") as fd:
-            if not xpath:
-                ret = fd.read().strip()
-            else:
-                ret = _read_typed_metric(typ, xpath, fd)
-    # Json path library has to be replaced or wrapped in
-    # order to fix this too broad except clause.
-    except Exception:
-        logger.warning(
-            "unable to read metric in '{}' in branch '{}'".format(
-                path, branch
-            ),
-            parse_exception=True,
-        )
-
-    return ret
+    if xpath:
+        try:
+            return _read_typed_metric(typ, xpath.strip(), fd)
+        # Json path library has to be replaced or wrapped in
+        # order to fix this too broad except clause.
+        except Exception:
```

Comment:
there is a change in behavior - if path is a directory or is missing in one of the branches we would have continued before. Now it fails. We should keep the scope for this except the same as possible. May be put a comment here that this scope is important.

Reply:
Would it be right to move the outer `if` block back inside the `try/except`?